src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -42,13 +42,17 @@ class DoctrineChoiceLoader implements ChoiceLoaderInterface
      *
      * @param ObjectManager              $manager      The object manager
      * @param string                     $class        The class name of the loaded objects
-     * @param IdReader                   $idReader     The reader for the object IDs
+     * @param IdReader|null              $idReader     The reader for the object IDs
      * @param EntityLoaderInterface|null $objectLoader The objects loader
      */
     public function __construct(ObjectManager $manager, string $class, IdReader $idReader = null, EntityLoaderInterface $objectLoader = null)
     {
         $classMetadata = $manager->getClassMetadata($class);
 
+        if ($idReader && !$idReader->isSingleId()) {
+            @trigger_error(sprintf('Passing an instance of "%s" with an entity class "%s" that have a composite id is deprecated since 4.3 and will throw an exception in 5.0.', IdReader::class, $class), E_USER_DEPRECATED);
```

Comment:
We should also add a commented line of PHP code that would just need to be enabled on 5.0.